src/main/java/io/vertx/httpproxy/interceptors/impl/HeadersInterceptorImpl.java

```diff
@@ -11,10 +11,50 @@
 
 package io.vertx.httpproxy.interceptors.impl;
 
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.httpproxy.ProxyContext;
 import io.vertx.httpproxy.ProxyInterceptor;
+import io.vertx.httpproxy.ProxyRequest;
+import io.vertx.httpproxy.ProxyResponse;
 
+import java.util.Set;
+
+/**
+ * The general interceptor for headers. Extended by other implementations.
+ */
 public class HeadersInterceptorImpl implements ProxyInterceptor {
+  Handler<MultiMap> changeRequestHeaders;
+  Handler<MultiMap> changeResponseHeaders;
+
+  public HeadersInterceptorImpl(Handler<MultiMap> changeRequestHeaders, Handler<MultiMap> changeResponseHeaders) {
+    this.changeRequestHeaders = changeRequestHeaders;
+    this.changeResponseHeaders = changeResponseHeaders;
+  }
+
+  public static HeadersInterceptorImpl filter(Set<CharSequence> requestHeaders, Set<CharSequence> responseHeaders) {
+    return new HeadersInterceptorImpl(oldRequestHeader -> {
+        if (requestHeaders == null) return;
```

Comment:
Since the set is known in advance, you could use a (constant) no-op handler when the set is empty